objectivec/GPBDescriptor.h

```diff
@@ -81,6 +81,16 @@ typedef NS_ENUM(uint8_t, GPBFieldType) {
 @property(nonatomic, readonly, getter=isWireFormat) BOOL wireFormat;
 /** The class of this message. */
 @property(nonatomic, readonly) Class messageClass;
+/**
+ * If this message was nested under another message, it will return
+ * that containing descriptor.
+ */
+@property(readonly, nullable) GPBDescriptor *containingType;
+/**
+ * Returns the full name of this message class (package.message). Can return
+ * nil if the runtime is unable to compute the value for some reason.
+ */
+@property(readonly, nullable) NSString *fullName;
```

Comment:
/** Fully qualified name for this message (package.message). Can be nil if the value is unable to be computed. */